packages/core/src/actions/fetchData.js

```diff
@@ -59,4 +82,29 @@ function fetchData(key, url, format = d => d, config = {}, useCache = true) {
 
 }
 
-export {fetchData};
+/** Launch the generic wrapper */
+async function launchWorkerWrapper(fn, callback) {
+  if (wrapperWorker) {
+    await wrapperWorker(proxy(fn), proxy(callback));
+  }
+  else {
+    callback(fn());
+  }
+}
+
+/** Generic webworker wrapper. */
+export function needWebWorkerPromise(key, fn) {
+  return new Promise((resolve, reject) => {
+    const callback = data => {
+      resolve({key, data});
+    };
+    try {
+      launchWorkerWrapper(fn, callback);
+    }
+    catch (error) {
+      console.error(error);
+      reject(error);
+    }
+  });
+}
+
```

Comment:
Is this function essentially just a `need` without any axios get? I don't necessarily see the use for this... but I DO see the need for a helper function that simply runs the supplied function and returns a promise (basically this, but without all the `key` and store stuff for needs). What do you think about:
1. simplifying this to just run the supplied function arguement in a web worker
2. moving it to a separate file within `src/helpers`